sympy/utilities/_compilation/tests/test_compilation.py

```diff
@@ -60,3 +64,23 @@ def test_compile_link_import_strings():
     finally:
         if info and info['build_dir']:
             shutil.rmtree(info['build_dir'])
+
+
+def test_compile_sources():
+    build_dir = tempfile.mkdtemp("sympy_test_compilation")
+    def _cleanup():
+        shutil.rmtree(build_dir)
+    atexit.register(_cleanup)
```

Comment:
It would be better to use `try/finally` rather than `atexit`.

Actually `tempfile.TemporaryDirectory` provides a context manager wrapping `mkdtemp`. Can that be used?

Reply:
I've had trouble with try/finally when using `--pdb` under `pytest`. But maybe that's no longer an issue?

I can change this to use the context manager from TemporaryDirectory.